### ISSUE TITLE
Delete-and-set single uid in the same transaction

### ISSUE THREAD

**Issue by wiradikusuma** [dgraph-io/dgraph#5298]
### What version of Dgraph are you using?

v1.2.2.

### Have you tried reproducing the issue with the latest release?

No, but I think my issue is related to (duplicate of?) https://github.com/dgraph-io/dgraph/issues/5017.

### What is the hardware spec (RAM, OS)?

Tried on both Windows and Ubuntu.

### Steps to reproduce the issue (command/config used to run Dgraph).

I have model `Blog` that contains `author` and `tags`, both are optional:

	<author>: uid .
	<tags>: [uid] .

If I want to update the `tags`, I can just empty it and repopulate it, both in the *same transaction*:

	delete {
	    <0xBLOG> <tags> * .
	}

	set {
	    ...
	}

If I use that approach to update the `author` (let say changing JOHN to MARY), I will end up with invalid data:

    {
      "uid": "0xBLOG",
      "author": {
        "uid": [
          "0xJOHN",
          "0xMARY"
        ]
      }
    }

If I split them (delete and set) to _different transactions_, it's working as expected.

### Expected behaviour and actual result.

Delete-and-set single uid in the same transaction should work.

**Comment by parasssh** [comment]
Could you provide more details on the issue and how to reproduce it? It may help to provide your full schema and the working and not working case.

**Comment by martinmr** [comment]
Closing this as a duplicate of https://github.com/dgraph-io/dgraph/issues/5017

@wiradikusuma Can you see the comment I left in the other thread? Using a single set instead of delete + set should work and get you unblocked. Dgraph understands that the author predicate is a single uid and the set will delete the old value. We'll look into the issue you reported. Thanks.
